sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -654,7 +660,313 @@ void setAndGetSessionState(MessagingEntityType entityType) {
+            .assertNext(message -> {
+                assertMessageEquals(message, messageId, isSessionEnabled);
+                messagesPending.decrementAndGet();
+            })
+            .assertNext(message -> {
+                assertMessageEquals(message, messageId, isSessionEnabled);
+                messagesPending.decrementAndGet();
+            })
+            .thenCancel()
+            .verify();
+    }
+
+    /**
+     * Verifies that we can renew session lock on a session receiver with multiple receiver.
+     */
+    @MethodSource("messagingEntityProvider")
+    @ParameterizedTest
+    void multipleReceiverRenewSessionLock(MessagingEntityType entityType) {
+        // Arrange
+        int firstBatchsize = 1;
+        int secondBatchSize = 1;
+
+        setSenderAndReceiver(entityType, true);
+
+        final String messageId = UUID.randomUUID().toString();
+        final List<ServiceBusMessage> messages = TestUtils.getServiceBusMessages(firstBatchsize, CONTENTS_BYTES,
+            messageId, sessionId);
+
+        messages.addAll(TestUtils.getServiceBusMessages(firstBatchsize, CONTENTS_BYTES, messageId, sessionId2));
+
+
+        final ReceiveAsyncOptions options = new ReceiveAsyncOptions()
+            .setIsAutoCompleteEnabled(false);
+
+        // Blocking here because it is not part of the scenario we want to test.
+        sendMessage(messages).block(TIMEOUT);
+
+        ServiceBusReceivedMessageContext receivedMessage = receiver.receive(options).next().block(TIMEOUT);
+        assertNotNull(receivedMessage.getMessage());
+
+        logger.info("Received message. Seq: {}.", receivedMessage.getMessage().getSequenceNumber());
+
+        // Assert & Act
+        try {
+            ServiceBusReceivedMessageContext finalReceivedMessage1 = receivedMessage;
+            StepVerifier.create(Mono.delay(Duration.ofSeconds(10))
+                .then(Mono.defer(() -> receiver.renewSessionLock(finalReceivedMessage1.getSessionId()))))
+                .assertNext(lockedUntil -> {
+                    assertNotNull(lockedUntil, "Could not renew session lock.");
+                })
+                .verifyComplete();
+        } finally {
+            logger.info("Completing message. Seq: {}.", receivedMessage.getMessage().getSequenceNumber());
+
+            receiver.complete(receivedMessage.getMessage())
+                .doOnSuccess(aVoid -> messagesPending.decrementAndGet())
+                .block(TIMEOUT);
+        }
+
+        receivedMessage = receiver2.receive(options).next().block(TIMEOUT);
+        assertNotNull(receivedMessage);
+
+        try {
+            ServiceBusReceivedMessageContext finalReceivedMessage = receivedMessage;
+            StepVerifier.create(Mono.delay(Duration.ofSeconds(10))
+                .then(Mono.defer(() -> receiver2.renewSessionLock(finalReceivedMessage.getSessionId()))))
+                .assertNext(lockedUntil -> {
+                    assertNotNull(lockedUntil, "Could not renew session lock.");
```

Comment:
this assertion is always true. There are other StepVerifier operators you can use.